### PR TITLE
Remove init-genesis-state command

### PR DESCRIPTION
**Pull request by karthikiyer56**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've reviewed the changes in this PR and if I consider them worthwhile for being mentioned on release notes then I have updated the relevant `CHANGELOG.md` within the  component folder structure. For example, if I changed horizon, then I updated ([services/horizon/CHANGELOG.md](services/horizon/CHANGELOG.md). I add a new line item describing the change and reference to this PR. If I don't update a CHANGELOG, I acknowledge this PR's change may not be mentioned in future release notes.  
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

This PR is related to issue #5469
The `init-genesis-state` functionality and all the accompanying logic and tests are removed as a part of this PR

### Why
Refer #5469 for more details

### CODE REVIEW

File: services/horizon/internal/ingest/fsm.go [hunk 1]
```diff
@@ -308,7 +308,7 @@ func (b buildState) run(s *system) (transition, error) {
 	// won't need bucket list hash and protocol version.
```
Comment (tamirms) [inline services/horizon/internal/ingest/fsm.go:308]: this comment should be updated

File: services/horizon/cmd/ingest.go [hunk 2]
```diff
@@ -259,56 +259,6 @@ var ingestTriggerStateRebuildCmd = &cobra.Command{
...
-		}
-
-		log.Info("Genesis ledger stat successfully ingested!")
-		return nil
-	},
-}
-
 var ingestBuildStateCmd = &cobra.Command{
 	Use:   "build-state",
 	Short: "builds state at a given checkpoint. warning! requires clean DB.",
```
Comment (tamirms) [inline services/horizon/cmd/ingest.go:264]: you can remove the `ingestBuildStateSequence != 1` condition from the checkpoint ledger validation in this command

File: services/horizon/internal/ingest/processor_runner_test.go [hunk 3]
```diff
@@ -79,7 +32,6 @@ func TestProcessorRunnerRunHistoryArchiveIngestionHistoryArchive(t *testing.T) {
 	defer mock.AssertExpectationsForObjects(t, historyAdapter)
 
 	m := &ingest.MockChangeReader{}
-	m.On("Read").Return(ingest.GenesisChange(network.PublicNetworkPassphrase), nil).Once()
```
Comment (tamirms) [inline services/horizon/internal/ingest/processor_runner_test.go:82]: the purpose of this test is to exercise the history archive ingestion. the test sets up a fake history archive consisting of one ledger entry. If you remove `m.On("Read").Return(ingest.GenesisChange(network.PublicNetworkPassphrase), nil).Once()` then this changes the test so that the history archive does not contain anything. I suggest replacing `ingest.GenesisChange(network.PublicNetworkPassphrase)` with another ledger entry so that the test can exercise ingesting from a non-empty history archive

Comment (karthikiyer56) [reply services/horizon/internal/ingest/processor_runner_test.go:82]: Yes, that question did come to my mind. 
I had meant to comment and tag you about that, except you beat me to it 😸 

Erm, may i ask you to point me to a fixture in another test, from where i can copy-pasta a ledger entry?

File: services/horizon/internal/ingest/fsm.go (hunk 1 above)
Comment (karthikiyer56) [reply services/horizon/internal/ingest/fsm.go:308]: I had debated removng the snippet `	// In the long term we should probably create artificial xdr.LedgerCloseMeta
	// for ledger #1 instead of using `ingest.GenesisChange` reader in
	// ProcessorRunner.RunHistoryArchiveIngestion().`

Except, i wasn't sure what to replace it with. 

Would you recommend simply removing the above highlighted snippet?

Comment (tamirms) [reply services/horizon/internal/ingest/fsm.go:308]: you can remove the snippet since we're no longer handling the gensis ledger as a special case

File: services/horizon/internal/ingest/processor_runner_test.go (hunk 3 above)
Comment (tamirms) [reply services/horizon/internal/ingest/processor_runner_test.go:82]: you can find some examples in services/horizon/internal/ingest/verify_range_state_test.go . There are also a bunch in the tests in the processors package:

services/horizon/internal/ingest/processors

File: services/horizon/internal/ingest/fsm.go [hunk 4]
```diff
@@ -301,9 +301,6 @@ func (b buildState) run(s *system) (transition, error) {
 
 	// We don't need to prepare range for genesis checkpoint because we don't
 	// perform protocol version and bucket list hash checks.
```
Comment (tamirms) [inline services/horizon/internal/ingest/fsm.go:303]: this block should also be deleted

Comment (karthikiyer56) [reply services/horizon/internal/ingest/fsm.go]: done

File: services/horizon/internal/ingest/processor_runner_test.go (hunk 3 above)
Comment (karthikiyer56) [reply services/horizon/internal/ingest/processor_runner_test.go:82]: done, updated the test fixture

File: services/horizon/internal/ingest/fsm.go (hunk 1 above)
Comment (karthikiyer56) [reply services/horizon/internal/ingest/fsm.go:303]: done

File: services/horizon/internal/ingest/fsm.go (hunk 4 above)
Comment (karthikiyer56) [reply services/horizon/internal/ingest/fsm.go]: done
